.github/workflows/test.yml

```diff
@@ -86,9 +86,10 @@ jobs:
         NUM_CORES: ${{ matrix.cores }}
 
     - name: Upload test results
-      # We always want the test results to be uploaded, even when cancelled.
+      # If the job was not cancelled, we want to save the result (this includes
+      # when the job fails). See warning here:
       # https://docs.github.com/en/actions/learn-github-actions/expressions#always
-      if: ${{ always() }}
+      if: ${{ !cancelled() }}
```

Comment:
Let's say vtr_reg_strong fails after 30 minutes, and I push a new commit to fix the failure (or I cancell the workflow manually). Is the artifact for the failed vtr_reg_strong uploaded?

Reply:
@soheilshahrouz Thats a great question! I just tested it! Each run (i.e. vtr_reg_strong for example) runs independently of each other. If this run fails it will generate its artifact and that will exist until its deleted (right now thats 90 days from now). If you push another commit, or cancel the run, this artifact will still exist but any test that did not complete yet (neither failed nor succeeded) will not generate an artifact.

Right now when you cancel a run part-way through execution it will stop running the test, then try to generate the artifacts; which in my opinion is a bit strange since the run did not complete.

Reply:
Here is a CI run for this PR which I cancelled which demonstrates what I have changed:
https://github.com/verilog-to-routing/vtr-verilog-to-routing/actions/runs/9421995802

As you can see vtr_reg_strong's (funny enough) artifact was generated and saved; but the tests that did not complete's aritfacts were not saved.